File: examples/flutter/lib/src/models/collection.dart
part of appwrite.models;

/// Collection
class Collection {
    /// Collection ID.
    final String $id;
    /// Collection permissions.
    final Permissions $permissions;
    /// Collection name.
    final String name;
    /// Collection creation date in Unix timestamp.
    final int dateCreated;
    /// Collection creation date in Unix timestamp.
    final int dateUpdated;
    /// Collection rules.
    final List<Rule> rules;

    Collection({
        required this.$id,
        required this.$permissions,
        required this.name,
        required this.dateCreated,
        required this.dateUpdated,
        required this.rules,
    });

    factory Collection.fromMap(Map<String, dynamic> map) {
        return Collection(
            $id: map['\$id'],
            $permissions: Permissions.fromMap(map['\$permissions']),
            name: map['name'],
            dateCreated: map['dateCreated'],
            dateUpdated: map['dateUpdated'],
            rules: List<Rule>.from(map['rules'].map((p) => Rule.fromMap(p))),
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "\$id": $id,
            "\$permissions": $permissions.toMap(),
            "name": name,
            "dateCreated": dateCreated,
            "dateUpdated": dateUpdated,
            "rules": rules.map((p) => p.toMap()),
        };
    }
}

File: examples/swift-server/Sources/AppwriteModels/Collection.swift

/// Collection
public class Collection {

    /// Collection ID.
    public let id: String

    /// Collection permissions.
    public let permissions: Permissions

    /// Collection name.
    public let name: String

    /// Collection creation date in Unix timestamp.
    public let dateCreated: Int

    /// Collection creation date in Unix timestamp.
    public let dateUpdated: Int

    /// Collection rules.
    public let rules: [Rule]

    init(
        id: String,
        permissions: Permissions,
        name: String,
        dateCreated: Int,
        dateUpdated: Int,
        rules: [Rule]
    ) {
        self.id = id
        self.permissions = permissions
        self.name = name
        self.dateCreated = dateCreated
        self.dateUpdated = dateUpdated
        self.rules = rules
    }

    public static func from(map: [String: Any]) -> Collection {
        return Collection(
            id: map["$id"] as! String,
            permissions: Permissions.from(map: map["$permissions"] as! [String: Any]),
            name: map["name"] as! String,
            dateCreated: map["dateCreated"] as! Int,
            dateUpdated: map["dateUpdated"] as! Int,
            rules: (map["rules"] as! [[String: Any]]).map { Rule.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "$id": id as Any,
            "$permissions": permissions.toMap() as Any,
            "name": name as Any,
            "dateCreated": dateCreated as Any,
            "dateUpdated": dateUpdated as Any,
            "rules": rules.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                                                                                                                                                                                            
}